packages/abi-coder/src/coders/enum.ts

```diff
@@ -67,7 +69,7 @@ export default class EnumCoder<TCoders extends Record<string, Coder>> extends Co
     const encodedValue = valueCoder.encode(value[caseKey]);
 
     const padding = new Uint8Array(this.#encodedValueSize - valueCoder.encodedLength);
-    return concat([this.#caseIndexCoder.encode(caseIndex), padding, encodedValue]);
+    return concatWithVectorData([this.#caseIndexCoder.encode(caseIndex), padding, encodedValue]);
```

Comment:
The mention of vectors (`concatWithVectorData`) inside of array/enum/struct/tuple coders feels a bit weird to me because they shouldn't be concerned with any ABI type except their own that they're decoding. 
The Rust SDK has the concept of [Data](https://github.com/FuelLabs/fuels-rs/blob/b731e9468ce39972a4eef3af86514768aee91f20/packages/fuels-core/src/types/unresolved_bytes.rs), a nice abstraction via which they've managed to keep the "known already from ABI how it'll be encoded" types like u8, b256, etc. separate from vector-related encoding by having the `Data.Inline` and `Data.Dynamic` enum values. The `Inline` values are in essence already known when their coder returns the encoded values, whereas the `Dynamic` has all the encoding figured out except for `RawVec`'s pointer value, which is recursively set via the `.resolve()` method wherever the encoding needs to happen. By doing it like this, all their coders have a single responsibility of encoding their specific type, and their implementations are vector-agnostic, so to speak.

As far as I'm concerned, this PR looks good - everything works and we should merge it, but I'd love to have a further discussion on the design decisions at some later time.